Examples/PassGH/spinLockAtomicBranch.cvf

```diff
@@ -0,0 +1,62 @@
+// Allocatable version of Examples/Pass/spinLock.cvf.
+
+typedef int Lock;
+
+pragma grasshopper_include {./grasshopper/spinlock-module.spl};
+pragma grasshopper_footprint {LockFoot};
+pragma grasshopper_footprint_sort {Set<Lock>};
+
+thread Lock x, ret;
+thread bool test;  // Used when trying to take the lock.
+
+view newLock(Lock x), holdLock(Lock x), isLock(Lock x);
+
+method newLock() {
+  {| emp |}
+    <| ret = %{new Lock}; |>
+  {| newLock(ret) |}
+    <| %{[|ret|].lock := false}; |>
+  {| isLock(ret) |}
+}
+
+method lock(Lock x) {
+  {| isLock(x) |}
+    do {
+      {| isLock(x) |}
+        test = false;
+      {| if test == false then isLock(x) else False() |}
+        <| test = %{ [|x|].lock };
+           if (%{ [|x|].lock } == false) {
```

Comment:
As before, this is just a variant of the allocatable spinlock using the new atomic branch syntax.